diff --git a/boot/config.txt b/boot/config.txt
index 41b922f..4612553 100644
--- a/boot/config.txt
+++ b/boot/config.txt
@@ -14,3 +14,4 @@ initramfs ramdisk.img 0x01f00000
 mask_gpu_interrupt0=0x400
 avoid_warnings=2
 enable_uart=1
+dtoverlay=sdio,poll_once=off
diff --git a/init.rpi3.rc b/init.rpi3.rc
index c37494f..ed06927 100644
--- a/init.rpi3.rc
+++ b/init.rpi3.rc
@@ -39,6 +39,9 @@ on boot
     setprop service.adb.tcp.port 5555
     start adbd
 
+    #load wlan-usb
+    insmod /lib/modules/wlan-usb.ko
+
 service wpa_supplicant /system/bin/wpa_supplicant \
     -iwlan0 -Dnl80211 -c/data/misc/wifi/wpa_supplicant.conf \
     -I/system/etc/wifi/wpa_supplicant_overlay.conf \
