diff --git a/include/linux/skbuff.h b/include/linux/skbuff.h
index d443d9ab0236..a0fa91daf14f 100644
--- a/include/linux/skbuff.h
+++ b/include/linux/skbuff.h
@@ -566,7 +566,7 @@ struct sk_buff {
 	 * want to keep them across layers you have to do a skb_clone()
 	 * first. This is owned by whoever has the skb queued ATM.
 	 */
-	char			cb[48] __aligned(8);
+	char			cb[72] __aligned(8);
 
 	unsigned long		_skb_refdst;
 	void			(*destructor)(struct sk_buff *skb);
